src/controllers/post.controller.js

```diff
@@ -1,99 +1,135 @@
+
+  if (description) {
+    tags = filterTags(description);
+  }
+
+  try {
+    const postId = await insertNewPost({ description, userId, url });
+
+    if (tags.length) {
+      await tags.forEach(async (tag) => {
+        let tagId = await getTag(tag);
+        if (!tagId?.rows.length) {
+          tagId = await insertNewTag(tag);
+        }
+
+        await insertNewTagQuote({
+          post: postId?.rows[0]?.id,
+          tag: tagId?.rows[0]?.id,
+        });
+      });
+    }
+
+    return res.sendStatus(201);
+  } catch (error) {
+    return res.sendStatus(500);
+  }
 }
 
 async function getTimelinePosts(req, res) {
-	try {
-		const posts = await getPosts();
-
-		const completePosts = await Promise.all(
-			posts.rows.map(async (post) => {
-				const { title, image, url, description } = await urlMetadata(post.url);
-				return {
-					...post,
-					url: {
-						title,
-						image,
-						path: url,
-						description,
-					},
-				};
-			})
-		);
-
-		res.status(200).send(completePosts);
-	} catch (error) {
-		return res.sendStatus(500);
-	}
+  try {
+    const posts = await getPosts();
+
+    const completePosts = await Promise.all(
+      posts.rows.map(async (post) => {
+        const { title, image, url, description } = await urlMetadata(post.url);
+        return {
+          ...post,
+          url: {
+            title,
+            image,
+            path: url,
+            description,
+          },
+        };
+      })
+    );
+
+    res.status(200).send(completePosts);
+  } catch (error) {
+    return res.sendStatus(500);
+  }
 }
 
 async function getLikes(req, res) {
-	const postId = req.params.postId;
-	try {
-		const likes = (await listLikes(postId)).rows;
-		return res.status(200).send(likes);
-	} catch {
-		return res.sendStatus(500);
-	}
+  const postId = req.params.postId;
+  try {
+    const likes = (await listLikes(postId)).rows;
+    return res.status(200).send(likes);
+  } catch {
+    return res.sendStatus(500);
+  }
 }
 async function changeLikes(req, res) {
-	const postId = req.params.postId;
-	const userId = res.locals.userId;	
-	try {
-		const like = (await getLikeByIds({ postId, userId })).rows;
-		if (like.length === 0) {
-			await insertLike({ postId, userId });
-			return res.sendStatus(201);
-		}
-		await deleteLike({ postId, userId });
-		return res.sendStatus(204);
-	} catch {
-		return res.sendStatus(500);
-	}
+  const postId = req.params.postId;
+  const userId = res.locals.userId;
+  try {
+    const like = (await getLikeByIds({ postId, userId })).rows;
+    if (like.length === 0) {
+      await insertLike({ postId, userId });
+      return res.sendStatus(201);
+    }
+    await deleteLike({ postId, userId });
+    return res.sendStatus(204);
+  } catch {
+    return res.sendStatus(500);
+  }
+}
+
+async function editPosts(req, res) {
+  const { description } = req.body;
+  const postId = req.params.postId;
+
+  if (!description || !postId) {
+    return res.sendStatus(404);
+  }
```

Comment:
Wouldn't it be error 422?

Reply:
In fact, we shouldn't validate   !description   here because it won't allow someone to delete a previous description. I already applied this fix.
For a post the doesn't exist, I think it is ok to respond with a Not Found. But, we can talk about this better.

Reply:
We also don't need to validate the postId...since it is been validating on postValidate Middleware. I'm removing it too.